### PR TITLE
updating pose via colorSensor theoritical

### PR DESCRIPTION
**Pull request by water221c**
Before issuing a pull request, please see the contributing page.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Sensors/OrbitColorSensor.java [hunk 1]
```diff
@@ -20,11 +25,15 @@ public static float[] rgb() {
...
+        public static String readColor(float red, float green, float blue) {
+            if ((red >= rgb()[0] + 10 || red <= rgb()[0] - 10) && (green >= rgb()[1] + 10)) {
+                return "red";
+            }
+            else if( ( green <= rgb()[1] - 10) && (blue >= rgb()[2] + 10 || blue <= rgb()[2] - 10)){
+                return "blue";
+            }
+            else
+                return "null";
         }
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Sensors/OrbitColorSensor.java:37]: What is this function for?

We already have rgb to read the colors..

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Sensors/OrbitDistSensor.java [hunk 2]
```diff
@@ -7,12 +7,12 @@
...
+    private static ModernRoboticsI2cRangeSensor sensor= null;
 
     public OrbitDistSensor(HardwareMap hardwareMap) {
         sensor = hardwareMap.get(ModernRoboticsI2cRangeSensor.class, "distanceSensor");
     }
 
-    public float getDistance(){return (float) sensor.getDistance(DistanceUnit.CM);}
+    public static float getDistance(){return (float) sensor.getDistance(DistanceUnit.CM);}
 
 }
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Sensors/OrbitDistSensor.java:18]: This class shouldn't be static - to use it create an instance of the class.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/positionTracker/PoseTracker.java [hunk 3]
```diff
@@ -38,7 +38,34 @@ public static Pose2d getPose() {
...
+                case blue:
+                    if (color == "blue")
+                        setPose(new Pose2d(xyAfterReset.x,xyAfterReset.y, getHeading()));
+                    else if (color == "red")
+                        setPose(new Pose2d(xyAfterReset.x + Constants.differenceInDisBetweenColors, xyAfterReset.y, getHeading()));
+                    break;
+            }
+
+        }
+        return;
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/positionTracker/PoseTracker.java:67]: Why is it here?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotData/Alliance.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package org.firstinspires.ftc.teamcode.robotData;
+
+public enum Alliance {
+    red,blue
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotData/Alliance.java:4]: Write in capital letters (RED, BLUE)

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotData/GlobalData.java [hunk 5]
```diff
@@ -7,6 +7,7 @@ public class GlobalData {
 
     public static RobotState robotState = RobotState.TRAVEL;
     public static boolean inAutonomous;
+    public static boolean blueOrRedStart_Red;
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotData/GlobalData.java:10]: Change this Boolean name to isRed

**Review by Ido-Sobol — CHANGES_REQUESTED** [review]

**Comment by Bar2005** [comment]
Closing this pr because we need to think what is the proper way to write it and where, we will think of it again as soon as we will start write autonomous

**Pull request closed without merging**